File: screener/criteria/plugins/intraday_momentum.py
"""Liquid movers with relative-volume surge and clean trend."""

from __future__ import annotations

from tradingview_screener import col

from screener.criteria import criterion


@criterion("intraday_momentum")
def intraday_momentum() -> list:
    """Designed for intraday trading: filters for above-average current volume
    vs. 10d average, today moving meaningfully, price riding above the
    short EMA, and RSI in trend-strong territory.
    """
    return [
        col("relative_volume_10d_calc") >= 1.5,
        col("volume") >= 200_000,
        col("close") >= col("EMA20"),
        col("EMA20") > col("EMA200"),
        col("RSI") >= 55,
        col("RSI") <= 80,
        col("change") >= 1.0,
    ]

File: screener-rs/src/screeners/criteria/intraday_momentum.rs
use super::{CmpOp, Predicate, cmp, field, val};

pub const NAME: &str = "intraday_momentum";

pub fn predicates() -> Vec<Predicate> {
    vec![
        cmp(field("relative_volume_10d_calc"), CmpOp::Gte, val(1.5)),
        cmp(field("volume"), CmpOp::Gte, val(200_000.0)),
        cmp(field("close"), CmpOp::Gte, field("EMA20")),
        cmp(field("EMA20"), CmpOp::Gt, field("EMA200")),
        cmp(field("RSI"), CmpOp::Gte, val(55.0)),
        cmp(field("RSI"), CmpOp::Lte, val(80.0)),
        cmp(field("change"), CmpOp::Gte, val(1.0)),
    ]
}
